### PR TITLE
Backward compatibility with old auth classes

### PR DESCRIPTION
**Pull request by jduo**

### CODE REVIEW

File: java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java [hunk 1]
```diff
@@ -51,30 +54,42 @@
...
+    this.authHandler = authHandler;
     this.executors = new ContextPropagatingExecutorService(executors);
   }
 
-  private FlightContext makeContext(ServerCallStreamObserver<?> responseObserver) {
+  private CallContext makeContext(ServerCallStreamObserver<?> responseObserver) {
+    // Try to get the peer identity from middleware first (using the auth2 interfaces).
     final ContextAdapter contextAdapter = new ContextAdapter();
-    return new FlightContext(contextAdapter.get(AuthConstants.PEER_IDENTITY_KEY), responseObserver::isCancelled);
+    String peerIdentity = contextAdapter.get(AuthConstants.PEER_IDENTITY_KEY);
```
Comment (keeratsingh) [inline java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java]: Can we use the fully qualified name here, to differentiate between auth and auth2 constants ?
`org.apache.arrow.flight.auth2.AuthConstants`

File: java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java [hunk 2]
```diff
@@ -51,30 +54,42 @@
...
+      peerIdentity = org.apache.arrow.flight.auth.AuthConstants.PEER_IDENTITY_KEY.get();
+    }
+
+    return new CallContext(peerIdentity, responseObserver::isCancelled);
   }
 
   @Override
   public StreamObserver<Flight.HandshakeRequest> handshake(StreamObserver<Flight.HandshakeResponse> responseObserver) {
-    return ServerHandshakeWrapper.wrapHandshake(responseObserver, executors);
+    // This method is not meaningful with the auth2 interfaces. Authentication would already
```
Comment (keeratsingh) [inline java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java:83]: Could this method be called by someone using Auth2 with headers? If so, how would that case be handled?

**Review by keeratsingh — APPROVED** [review]
Apart from a couple of comments, the rest of the code changes look good.

File: java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java (hunk 2 above)
Comment (jduo) [reply java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java:83]: Someone could call this with auth2 headers and if so, it'd run the auth1 workflow (which would likely be a no-op if the server was configured to use the NoOp auth handler)

File: java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java (hunk 1 above)
Comment (jduo) [reply java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java]: Did this the other way around. Renamed auth2.AuthConstants to auth2.Auth2Constants and remove the FQCN from the auth.AuthConstants reference.

**Pull request merged**